x-pack/plugins/alerting/server/rules_client/methods/bulk_enable.ts

```diff
@@ -89,18 +98,36 @@ export const bulkEnableRules = async (context: RulesClientContext, options: Bulk
     taskManager: context.taskManager,
   });
 
-  const updatedRules = rules.map(({ id, attributes, references }) => {
-    return getAlertFromRaw(
-      context,
-      id,
-      attributes.alertTypeId as string,
-      attributes as RawRule,
-      references,
-      false
+  const enabledRules = rules.map(({ id, attributes, references }) => {
+    // TODO (http-versioning): alertTypeId should never be null, but we need to
+    // fix the type cast from SavedObjectsBulkUpdateObject to SavedObjectsBulkUpdateObject
+    // when we are doing the bulk disable and this should fix itself
+    const ruleType = context.ruleTypeRegistry.get(attributes.alertTypeId!);
+    const ruleDomain = transformRuleAttributesToRuleDomain<Params>(
+      attributes as RuleAttributes,
+      {
+        id,
+        logger: context.logger,
+        ruleType,
+        references,
+        omitGeneratedValues: false,
+      },
+      (connectorId: string) => actionsClient.isSystemAction(connectorId)
     );
+
+    try {
+      ruleDomainSchema.validate(ruleDomain);
+    } catch (e) {
+      context.logger.warn(`Error validating bulk enabled rule domain object for id: ${id}, ${e}`);
+    }
+    return ruleDomain;
```

Comment:
We can avoid the two loop if just do that, correct?
```suggestion
    return transformRuleDomainToRule<Params>(ruleDomain);
```

Reply:
I've maid this change on `bulk enable` PR: [bb43e3fa3074c88c0a86899f3acae71873e973c2
](https://github.com/elastic/kibana/pull/170476/commits/bb43e3fa3074c88c0a86899f3acae71873e973c2)

It'll appear here after merge